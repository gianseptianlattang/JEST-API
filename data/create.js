const randomstring = require('randomstring');

export const request = () => ({
        name: 'morpheus',
        job: 'leader',
});

export const negativeCase = [
    {
        title: 'null on name field',
        request: {
          name: null,
        },
      },
      {
        title: 'empty string on name field',
        request: {
          name: '',
        },
      },
      {
        title: 'number on name field',
        request: {
          name: randomstring.generate({ charset: 'numeric', length: '10' }),
        },
      },
      {
        title: 'less than 3 char on name field',
        request: {
          name: randomstring.generate({ charset: 'alphabetic', length: '2' }),
        },
      },
      {
        title: 'greater than 30 char on name field',
        request: {
          name: randomstring.generate({ charset: 'alphabetic', length: '31' }),
        },
      },
      {
        title: 'symbol @ on name field',
        request: {
          name: `@${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol . on name field',
        request: {
          name: `.${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol , on name field',
        request: {
          name: `,${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol _ on name field',
        request: {
          name: `_${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol ! on name field',
        request: {
          name: `!${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol / on name field',
        request: {
          name: `/${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol ? on name field',
        request: {
          name: `?${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol # on name field',
        request: {
          name: `#${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol $ on name field',
        request: {
          name: `$${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
    
      {
        title: 'symbol % on name field',
        request: {
          name: `%${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol ^ on name field',
        request: {
          name: `^${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol & on name field',
        request: {
          name: `&${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol * on name field',
        request: {
          name: `*${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'null on job field',
        request: {
          job: null,
        },
      },
      {
        title: 'empty string on job field',
        request: {
          job: '',
        },
      },
      {
        title: 'number on job field',
        request: {
          name: randomstring.generate({ charset: 'numeric', length: '10' }),
        },
      },
      {
        title: 'less than 3 char on job field',
        request: {
          job: randomstring.generate({ charset: 'alphabetic', length: '2' }),
        },
      },
      {
        title: 'greater than 30 char on job field',
        request: {
          job: randomstring.generate({ charset: 'alphabetic', length: '31' }),
        },
      },
      {
        title: 'symbol @ on job field',
        request: {
          job: `@${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol . on job field',
        request: {
          job: `.${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol , on job field',
        request: {
          job: `,${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol _ on job field',
        request: {
          job: `_${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol ! on job field',
        request: {
          job: `!${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol / on job field',
        request: {
          job: `/${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol ? on job field',
        request: {
          job: `?${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol # on job field',
        request: {
          job: `#${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol $ on job field',
        request: {
          job: `$${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
    
      {
        title: 'symbol % on job field',
        request: {
          job: `%${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol ^ on job field',
        request: {
          job: `^${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol & on job field',
        request: {
          job: `&${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol * on job field',
        request: {
          job: `*${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
];

export const snapEx = {
    id: expect.any(String),
    createdAt: expect.any(String),
};