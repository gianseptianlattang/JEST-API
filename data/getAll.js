export const queryparam = () => ({
    page : 2
});

export const casequeryparam = [
    {
        title: 'page 1',
        request: {
          page: 1,
        },
    },
    {
        title: 'page 2',
        request: {
          page: 2,
        },
    },
];

//Untuk get biasanya cek negatif case token, namun reqres.in tidak pake token.
// export const negativeTokenRequest = [
//     {
//       desc: 'without token',
//       request: {
//         token: '',
//       },
//     },
//     {
//       desc: 'with null token',
//       request: {
//         token: null,
//       },
//     },
//     {
//       desc: 'with wrong token - Man1',
//       request: {
//         token: Man1,
//       },
//     },
//     {
//       desc: 'with wrong token - Admin1',
//       request: {
//         token: Admin1,
//       },
//     },
//     {
//       desc: 'with random token',
//       request: {
//         token: randomToken,
//       },
//     },
//   ];