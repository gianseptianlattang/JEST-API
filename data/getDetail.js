export const queryparam = () => ({
    id : 2
});

export const casequeryparamid = [
    {
        title: 'id 1',
        request: {
          id: 1,
        },
    },
    {
        title: 'id 2',
        request: {
          id: 2,
        },
    },
];

//Untuk get biasanya cek negatif case token, namun reqres.in tidak pake token.
// export const negativeTokenRequest = [
//     {
//       desc: 'without token',
//       request: {
//         token: '',
//       },
//     },
//     {
//       desc: 'with null token',
//       request: {
//         token: null,
//       },
//     },
//     {
//       desc: 'with wrong token - Man1',
//       request: {
//         token: Man1,
//       },
//     },
//     {
//       desc: 'with wrong token - Admin1',
//       request: {
//         token: Admin1,
//       },
//     },
//     {
//       desc: 'with random token',
//       request: {
//         token: randomToken,
//       },
//     },
//   ];