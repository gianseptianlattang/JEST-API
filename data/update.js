const randomstring = require('randomstring');

export const request = () => ({
        name: 'morpheus2',
        job: 'member',
});

//Untuk Update biasanya cek negatif case token, namun reqres.in tidak pake token.
// export const negativeTokenRequest = [
//     {
//       desc: 'without token',
//       request: {
//         token: '',
//       },
//     },
//     {
//       desc: 'with null token',
//       request: {
//         token: null,
//       },
//     },
//     {
//       desc: 'with wrong token - Man1',
//       request: {
//         token: Man1,
//       },
//     },
//     {
//       desc: 'with wrong token - Admin1',
//       request: {
//         token: Admin1,
//       },
//     },
//     {
//       desc: 'with random token',
//       request: {
//         token: randomToken,
//       },
//     },
//   ];

export const negativeCase = [
    {
        title: 'null on name field',
        request: {
          name: null,
        },
      },
      {
        title: 'empty string on name field',
        request: {
          name: '',
        },
      },
      {
        title: 'number on name field',
        request: {
          name: randomstring.generate({ charset: 'numeric', length: '10' }),
        },
      },
      {
        title: 'less than 3 char on name field',
        request: {
          name: randomstring.generate({ charset: 'alphabetic', length: '2' }),
        },
      },
      {
        title: 'greater than 30 char on name field',
        request: {
          name: randomstring.generate({ charset: 'alphabetic', length: '31' }),
        },
      },
      {
        title: 'symbol @ on name field',
        request: {
          name: `@${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol . on name field',
        request: {
          name: `.${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol , on name field',
        request: {
          name: `,${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol _ on name field',
        request: {
          name: `_${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol ! on name field',
        request: {
          name: `!${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol / on name field',
        request: {
          name: `/${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol ? on name field',
        request: {
          name: `?${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol # on name field',
        request: {
          name: `#${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol $ on name field',
        request: {
          name: `$${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
    
      {
        title: 'symbol % on name field',
        request: {
          name: `%${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol ^ on name field',
        request: {
          name: `^${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol & on name field',
        request: {
          name: `&${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol * on name field',
        request: {
          name: `*${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'null on job field',
        request: {
          job: null,
        },
      },
      {
        title: 'empty string on job field',
        request: {
          job: '',
        },
      },
      {
        title: 'number on job field',
        request: {
          name: randomstring.generate({ charset: 'numeric', length: '10' }),
        },
      },
      {
        title: 'less than 3 char on job field',
        request: {
          job: randomstring.generate({ charset: 'alphabetic', length: '2' }),
        },
      },
      {
        title: 'greater than 30 char on job field',
        request: {
          job: randomstring.generate({ charset: 'alphabetic', length: '31' }),
        },
      },
      {
        title: 'symbol @ on job field',
        request: {
          job: `@${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol . on job field',
        request: {
          job: `.${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol , on job field',
        request: {
          job: `,${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol _ on job field',
        request: {
          job: `_${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol ! on job field',
        request: {
          job: `!${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol / on job field',
        request: {
          job: `/${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol ? on job field',
        request: {
          job: `?${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol # on job field',
        request: {
          job: `#${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol $ on job field',
        request: {
          job: `$${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
    
      {
        title: 'symbol % on job field',
        request: {
          job: `%${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol ^ on job field',
        request: {
          job: `^${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol & on job field',
        request: {
          job: `&${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol * on job field',
        request: {
          job: `*${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
];

export const snapEx = {
    updatedAt: expect.any(String),
};