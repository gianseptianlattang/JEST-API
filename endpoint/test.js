const supertest = require('supertest');
require('dotenv').config();

const api = supertest('https://reqres.in');

export const create = (body) => api.post('/api/users')
  .set('Content-Type', 'application/json')
  .set('Accept', 'application/json')
  .send(body);

export const update = (body, id) => api.put(`/api/users/${id}`)
  .set('Content-Type', 'application/json')
  .set('Accept', 'application/json')
  .send(body);

  export const getAll = (queryparam) => api.get('/api/users?')
  .set('Content-Type', 'application/json')
  .set('Accept', 'application/json')
  .query(queryparam);

  export const getDetail = (id) => api.get(`/api/users/${id}`)
  .set('Content-Type', 'application/json')
  .set('Accept', 'application/json')

  export const deleteUser = (id) => api.delete(`/api/users/${id}`)
  .set('Content-Type', 'application/json')
  .set('Accept', 'application/json')