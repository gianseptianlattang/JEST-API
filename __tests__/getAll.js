import * as data from '../data/getAll';
import * as endpoint from '../endpoint/test';
// import * as schema from '../schema/create'; untuk jest terbaru tidak ada fungsi tomatchschema

let body;
let response;

describe('Get All User -- Positif Case', () => {
//   beforeAll(async () => {
//     await base.seedingData('');
//   });
    (data.casequeryparam).forEach(async (element) => {
        test(`Get All User on ${element.title}`, async () => {
            body = data.queryparam();
            const param = Object.keys(element.request);
            body[param]= element.request[param];
            response = await endpoint.getAll(body);
            expect(response.status).toEqual(200); 
            expect(response.body).toMatchSnapshot(); 
        });
    });
});