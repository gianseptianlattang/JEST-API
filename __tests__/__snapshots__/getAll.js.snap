// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get All User -- Positif Case Get All User on page 1 1`] = `
Object {
  "data": Array [
    Object {
      "avatar": "https://reqres.in/img/faces/1-image.jpg",
      "email": "george.bluth@reqres.in",
      "first_name": "George",
      "id": 1,
      "last_name": "Bluth",
    },
    Object {
      "avatar": "https://reqres.in/img/faces/2-image.jpg",
      "email": "janet.weaver@reqres.in",
      "first_name": "Janet",
      "id": 2,
      "last_name": "Weaver",
    },
    Object {
      "avatar": "https://reqres.in/img/faces/3-image.jpg",
      "email": "emma.wong@reqres.in",
      "first_name": "Emma",
      "id": 3,
      "last_name": "Wong",
    },
    Object {
      "avatar": "https://reqres.in/img/faces/4-image.jpg",
      "email": "eve.holt@reqres.in",
      "first_name": "Eve",
      "id": 4,
      "last_name": "Holt",
    },
    Object {
      "avatar": "https://reqres.in/img/faces/5-image.jpg",
      "email": "charles.morris@reqres.in",
      "first_name": "Charles",
      "id": 5,
      "last_name": "Morris",
    },
    Object {
      "avatar": "https://reqres.in/img/faces/6-image.jpg",
      "email": "tracey.ramos@reqres.in",
      "first_name": "Tracey",
      "id": 6,
      "last_name": "Ramos",
    },
  ],
  "page": 1,
  "per_page": 6,
  "support": Object {
    "text": "To keep ReqRes free, contributions towards server costs are appreciated!",
    "url": "https://reqres.in/#support-heading",
  },
  "total": 12,
  "total_pages": 2,
}
`;

exports[`Get All User -- Positif Case Get All User on page 2 1`] = `
Object {
  "data": Array [
    Object {
      "avatar": "https://reqres.in/img/faces/7-image.jpg",
      "email": "michael.lawson@reqres.in",
      "first_name": "Michael",
      "id": 7,
      "last_name": "Lawson",
    },
    Object {
      "avatar": "https://reqres.in/img/faces/8-image.jpg",
      "email": "lindsay.ferguson@reqres.in",
      "first_name": "Lindsay",
      "id": 8,
      "last_name": "Ferguson",
    },
    Object {
      "avatar": "https://reqres.in/img/faces/9-image.jpg",
      "email": "tobias.funke@reqres.in",
      "first_name": "Tobias",
      "id": 9,
      "last_name": "Funke",
    },
    Object {
      "avatar": "https://reqres.in/img/faces/10-image.jpg",
      "email": "byron.fields@reqres.in",
      "first_name": "Byron",
      "id": 10,
      "last_name": "Fields",
    },
    Object {
      "avatar": "https://reqres.in/img/faces/11-image.jpg",
      "email": "george.edwards@reqres.in",
      "first_name": "George",
      "id": 11,
      "last_name": "Edwards",
    },
    Object {
      "avatar": "https://reqres.in/img/faces/12-image.jpg",
      "email": "rachel.howell@reqres.in",
      "first_name": "Rachel",
      "id": 12,
      "last_name": "Howell",
    },
  ],
  "page": 2,
  "per_page": 6,
  "support": Object {
    "text": "To keep ReqRes free, contributions towards server costs are appreciated!",
    "url": "https://reqres.in/#support-heading",
  },
  "total": 12,
  "total_pages": 2,
}
`;
